wdl-lint/src/lib.rs

```diff
@@ -80,6 +80,17 @@ pub trait Rule: Visitor<State = Diagnostics> {
     ///
     /// If `None` is returned, all nodes are exceptable.
     fn exceptable_nodes(&self) -> Option<&'static [SyntaxKind]>;
+
+    /// Gets the ID of rules that are related to this rule.
+    ///
+    /// This can be used by tools (like `sprocket explain`) to suggest other
+    /// relevant rules to the user based on potential logical connections or
+    /// common co-occurrences of issues.
+    ///
+    /// By default, this returns an empty list.
+    fn related_rules(&self) -> &[&'static str] {
+        &[]
+    }
```

Comment:
```suggestion
    fn related_rules(&self) -> &[&'static str];
```
My thinking here is that requiring every rule to implement this will reduce maintenance burden. When someone is adding a new rule to this crate, we want them to think about what other rules are related, and forcing them to implement this is a way to achieve that.

The alternative, (leaving this with a default `impl`) allows new rules to ignore this method, meaning we might need to come back at a later date to add something which was overlooked at first.